### PR TITLE
Added test resources setup cleanup(copy)

### PR DESCRIPTION
**Pull request by mkoshurnikov**
Added:

- CreateTestResources (BQ Table, Gcs Bucket)
- RemoveTestResources(BQ Table, Gcs Bucket)
- resources: table schema json, table data json

### CODE REVIEW

**Comment by mkoshurnikov** [comment]
I'm working on it.

File: retail/interactive-tutorial/TestResourcesSetupCleanup/CreateTestResources.cs [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+{
+    private static readonly string ProductFileName = "products.json";
+    private static readonly string EventsFileName = "user_events.json";
+
+    private static readonly string ProductDataSet = "products";
+    private static readonly string ProductTable = "products";
+    private static readonly string EventsDataSet = "user_events";
+    private static readonly string EventsTable = "events";
+
+    private static readonly string productsBucketName = Environment.GetEnvironmentVariable("BUCKET_NAME");
```
Comment (amanda-tarafa) [inline retail/interactive-tutorial/TestResourcesSetupCleanup/CreateTestResources.cs:323]:
```suggestion
    private static readonly string productsBucketName = Environment.GetEnvironmentVariable("RETAIL_BUCKET_NAME");
```

File: retail/interactive-tutorial/TestResourcesSetupCleanup/CreateTestResources.cs [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+    private static readonly string eventsBucketName = Environment.GetEnvironmentVariable("EVENTS_BUCKET_NAME");
```
Comment (amanda-tarafa) [inline retail/interactive-tutorial/TestResourcesSetupCleanup/CreateTestResources.cs:324]:
```suggestion
    private static readonly string eventsBucketName = Environment.GetEnvironmentVariable("RETAIL_EVENTS_BUCKET_NAME");
```

File: retail/interactive-tutorial/TestResourcesSetupCleanup/RemoveTestResources.cs [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+/// </summary>
+public static class RemoveTestResourcesTutorial
+{
+    private const string ProductsDataSet = "products";
+    private const string EventsDataSet = "user_events";
+    private const string ProductsTable = "products";
+    private const string EventsTable = "events";
+
+    private static readonly string productsBucketName = Environment.GetEnvironmentVariable("BUCKET_NAME");
+    private static readonly string eventsBucketName = Environment.GetEnvironmentVariable("EVENTS_BUCKET_NAME");
```
Comment (amanda-tarafa) [inline retail/interactive-tutorial/TestResourcesSetupCleanup/RemoveTestResources.cs:167]:
```suggestion
    private static readonly string productsBucketName = Environment.GetEnvironmentVariable("RETAIL_BUCKET_NAME");
    private static readonly string eventsBucketName = Environment.GetEnvironmentVariable("RETAIL_EVENTS_BUCKET_NAME");
```

File: retail/interactive-tutorial/TestResourcesSetupCleanup/README.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
+# How to set up/ tear down the test resources
+
+## Required environment variables
+
+To successfully import the catalog data for tests, the following environment variables should be set:
+ - GOOGLE_CLOUD_PROJECT_NUMBER
```
Comment (amanda-tarafa) [inline retail/interactive-tutorial/TestResourcesSetupCleanup/README.md:6]: I didn't need this one.

File: retail/interactive-tutorial/TestResourcesSetupCleanup/README.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+ - GOOGLE_CLOUD_PROJECT_ID 
+ - BUCKET_NAME
+ - EVENTS_BUCKET_NAME
```
Comment (amanda-tarafa) [inline retail/interactive-tutorial/TestResourcesSetupCleanup/README.md:9]:
```suggestion
 - RETAIL_BUCKET_NAME
 - RETAIL_EVENTS_BUCKET_NAME
```

File: retail/interactive-tutorial/TestResourcesSetupCleanup/README.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+These values are stored in the Secret Manager and will be submitted as 
+   docker environment variables before the test run.
+
+The Secret Manager name is set in .kokoro/presubmit/common.cfg file, SECRET_MANAGER_KEYS variable.
```
Comment (amanda-tarafa) [inline retail/interactive-tutorial/TestResourcesSetupCleanup/README.md:14]: This informatio is not relevant for users.
```suggestion
```

File: retail/interactive-tutorial/TestResourcesSetupCleanup/README.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+## Import catalog data
+
+There is a JSON file with valid products prepared in the `TestResourcesSetupCleanup` directory:
+`TestResourcesSetupCleanup/resources/products.json`.
+
+Run the `CreateTestResources.cs` to perform the following actions:
+    - create the GCS bucket <BUCKET_NAME>,
```
Comment (amanda-tarafa) [inline retail/interactive-tutorial/TestResourcesSetupCleanup/README.md:22]: And the same for the rest.
```suggestion
    - create the GCS bucket <RETAIL_BUCKET_NAME>, 
```

File: retail/interactive-tutorial/TestResourcesSetupCleanup/RemoveTestResources.cs [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+    [Runner.Attributes.Example]
+    public static void PerformDeletionOfTestResources()
+    {
+        // Delete products and events GCS buckets
+        RemoveTestResources.DeleteBucket(productsBucketName);
+        RemoveTestResources.DeleteBucket(eventsBucketName);
+
+        // Delete all products from the Retail catalog
+        // DeleteAllProducts();
+
```
Comment (amanda-tarafa) [inline retail/interactive-tutorial/TestResourcesSetupCleanup/RemoveTestResources.cs:178]:
```suggestion
```

File: retail/interactive-tutorial/TestResourcesSetupCleanup/CreateTestResources.cs [hunk 9]
```diff
@@ -0,0 +1,357 @@
...
+        var buckets = storageClient.ListBuckets(projectId);
+
+        foreach (var bucket in buckets)
+        {
+            bucketsList.Add(bucket);
+            Console.WriteLine(bucket.Name);
+        }
+
+        return bucketsList;
+    }
```
Comment (amanda-tarafa) [inline retail/interactive-tutorial/TestResourcesSetupCleanup/CreateTestResources.cs:114]:
```suggestion
    public static IEnumerable<Bucket> ListBuckets() =>
        storageClient.ListBuckets(projectId);
```

File: retail/interactive-tutorial/TestResourcesSetupCleanup/CreateTestResources.cs [hunk 10]
```diff
@@ -0,0 +1,357 @@
...
+
+    /// <summary>Upload blob.</summary>
+    public static void UploadBlob(string bucketName, string localPath, string objectName)
+    {
+        var bucket = storageClient.GetBucket(bucketName);
+
```
Comment (amanda-tarafa) [inline retail/interactive-tutorial/TestResourcesSetupCleanup/CreateTestResources.cs:120]:
```suggestion
```

File: retail/interactive-tutorial/TestResourcesSetupCleanup/CreateTestResources.cs [hunk 11]
```diff
@@ -0,0 +1,357 @@
...
+            GcsPrefix = gcsErrorsBucket
+        };
+
+        string locationId = "global";
+        string catalogId = "default_catalog";
+        string branchId = "0";
+        BranchName defaultBranch = new BranchName(projectId, locationId, catalogId, branchId);
+
+        // To check error handling paste the invalid catalog name here:
+        // catalogId = "invalid_catalog_name"
```
Comment (amanda-tarafa) [inline retail/interactive-tutorial/TestResourcesSetupCleanup/CreateTestResources.cs:153]:
```suggestion

```

File: retail/interactive-tutorial/TestResourcesSetupCleanup/RemoveTestResources.cs [hunk 12]
```diff
@@ -0,0 +1,190 @@
...
+    /// <summary>Delete bucket.</summary>
+    public static void DeleteBucket(string bucketName)
+    {
+        Console.WriteLine($"Deleting Bucket {bucketName}.");
+
+        try
+        {
+            var bucketToDelete = storageClient.GetBucket(bucketName);
+            DeleteObjectsFromBucket(bucketToDelete);
+            storageClient.DeleteBucket(bucketToDelete.Name);
```
Comment (amanda-tarafa) [inline retail/interactive-tutorial/TestResourcesSetupCleanup/RemoveTestResources.cs:41]:
```suggestion
            storageClient.DeleteBucket(bucketName, new DeleteBucketOptions { DeleteObjects = true});
```

File: retail/interactive-tutorial/TestResourcesSetupCleanup/RemoveTestResources.cs [hunk 13]
```diff
@@ -0,0 +1,190 @@
...
+        Console.WriteLine($"Deleting object from bucket {bucket.Name}.");
+
+        var blobs = storageClient.ListObjects(bucket.Name);
+        foreach (var blob in blobs)
+        {
+            storageClient.DeleteObject(bucket.Name, blob.Name);
+        }
+
+        Console.WriteLine($"All objects are deleted from a GCS bucket {bucket.Name}.");
+    }
```
Comment (amanda-tarafa) [inline retail/interactive-tutorial/TestResourcesSetupCleanup/RemoveTestResources.cs:62]:
```suggestion
```

File: retail/interactive-tutorial/TestResourcesSetupCleanup/RemoveTestResources.cs [hunk 14]
```diff
@@ -0,0 +1,190 @@
...
+            Console.WriteLine($"Dataset {datasetId} and contents were deleted.");
+        }
+        catch (Exception)
+        {
+            Console.WriteLine($"Dataset {datasetId} does not exist.");
+        }
+    }
+
+    /// <summary>Delete Big Query dataset with tables.</summary>
+    public static void DeleteBQTable(string datasetId, string tableId)
```
Comment (amanda-tarafa) [inline retail/interactive-tutorial/TestResourcesSetupCleanup/RemoveTestResources.cs:133]: If you delete the dataset with its contents, you don't need to delete the tables individually. You can remore this method.

File: retail/interactive-tutorial/TestResourcesSetupCleanup/RemoveTestResources.cs [hunk 15]
```diff
@@ -0,0 +1,190 @@
...
+        // Delete products and events GCS buckets
+        RemoveTestResources.DeleteBucket(productsBucketName);
+        RemoveTestResources.DeleteBucket(eventsBucketName);
+
+        // Delete all products from the Retail catalog
+        // DeleteAllProducts();
+
+        // Delete events and products BQ tables
+        RemoveTestResources.DeleteBQTable(ProductsDataSet, ProductsTable);
+        RemoveTestResources.DeleteBQTable(EventsDataSet, EventsTable);
```
Comment (amanda-tarafa) [inline retail/interactive-tutorial/TestResourcesSetupCleanup/RemoveTestResources.cs:181]:
```suggestion
```

**Review by amanda-tarafa — CHANGES_REQUESTED** [review]
A few changes, but nothing major.

File: retail/interactive-tutorial/TestResourcesSetupCleanup/CreateTestResources.cs (hunk 1 above)
Comment (mkoshurnikov) [reply retail/interactive-tutorial/TestResourcesSetupCleanup/CreateTestResources.cs:323]: but in all tutorials just BUCKET_NAME without RETAIL

Comment (amanda-tarafa) [reply retail/interactive-tutorial/TestResourcesSetupCleanup/CreateTestResources.cs:323]: We have some other samples on CI (not Retail) using the BUCKET_NAME environment variable so Retail cannot use it. This was agreed months ago with @t-karasova and `RETAIL_BUCKET_NAME` and `RETAIL_EVENTS_BUCKET_NAME` are the env variables we created on CI a while back, and everything has been working fine.

Do the tutorials texts refere to the env variables, etc.? When we agreed to this it wasn't a problem.

Comment (mkoshurnikov) [reply retail/interactive-tutorial/TestResourcesSetupCleanup/CreateTestResources.cs:323]: yes, in all tutorials we have step when we create bucket and then export bucket name into this env variable: `export BUCKET_NAME=<YOUR_BUCKET_NAME>` and export BUCKET_NAME=<YOUR_BUCKET_NAME>

Comment (amanda-tarafa) [reply retail/interactive-tutorial/TestResourcesSetupCleanup/CreateTestResources.cs:323]: I've sent an internal email about this.

Comment (mkoshurnikov) [reply retail/interactive-tutorial/TestResourcesSetupCleanup/CreateTestResources.cs]: done

File: retail/interactive-tutorial/TestResourcesSetupCleanup/CreateTestResources.cs (hunk 2 above)
Comment (mkoshurnikov) [reply retail/interactive-tutorial/TestResourcesSetupCleanup/CreateTestResources.cs]: done

File: retail/interactive-tutorial/TestResourcesSetupCleanup/RemoveTestResources.cs (hunk 3 above)
Comment (mkoshurnikov) [reply retail/interactive-tutorial/TestResourcesSetupCleanup/RemoveTestResources.cs]: done

File: retail/interactive-tutorial/TestResourcesSetupCleanup/README.md (hunk 4 above)
Comment (mkoshurnikov) [reply retail/interactive-tutorial/TestResourcesSetupCleanup/README.md]: done

File: retail/interactive-tutorial/TestResourcesSetupCleanup/README.md (hunk 5 above)
Comment (mkoshurnikov) [reply retail/interactive-tutorial/TestResourcesSetupCleanup/README.md]: done

File: retail/interactive-tutorial/TestResourcesSetupCleanup/README.md (hunk 6 above)
Comment (mkoshurnikov) [reply retail/interactive-tutorial/TestResourcesSetupCleanup/README.md]: done

File: retail/interactive-tutorial/TestResourcesSetupCleanup/README.md (hunk 7 above)
Comment (mkoshurnikov) [reply retail/interactive-tutorial/TestResourcesSetupCleanup/README.md]: done

File: retail/interactive-tutorial/TestResourcesSetupCleanup/RemoveTestResources.cs (hunk 8 above)
Comment (mkoshurnikov) [reply retail/interactive-tutorial/TestResourcesSetupCleanup/RemoveTestResources.cs]: done

File: retail/interactive-tutorial/TestResourcesSetupCleanup/CreateTestResources.cs (hunk 9 above)
Comment (mkoshurnikov) [reply retail/interactive-tutorial/TestResourcesSetupCleanup/CreateTestResources.cs]: done

File: retail/interactive-tutorial/TestResourcesSetupCleanup/CreateTestResources.cs (hunk 10 above)
Comment (mkoshurnikov) [reply retail/interactive-tutorial/TestResourcesSetupCleanup/CreateTestResources.cs]: done

File: retail/interactive-tutorial/TestResourcesSetupCleanup/CreateTestResources.cs (hunk 11 above)
Comment (mkoshurnikov) [reply retail/interactive-tutorial/TestResourcesSetupCleanup/CreateTestResources.cs]: done

File: retail/interactive-tutorial/TestResourcesSetupCleanup/RemoveTestResources.cs (hunk 12 above)
Comment (mkoshurnikov) [reply retail/interactive-tutorial/TestResourcesSetupCleanup/RemoveTestResources.cs]: done

File: retail/interactive-tutorial/TestResourcesSetupCleanup/RemoveTestResources.cs (hunk 13 above)
Comment (mkoshurnikov) [reply retail/interactive-tutorial/TestResourcesSetupCleanup/RemoveTestResources.cs]: done

File: retail/interactive-tutorial/TestResourcesSetupCleanup/RemoveTestResources.cs (hunk 14 above)
Comment (mkoshurnikov) [reply retail/interactive-tutorial/TestResourcesSetupCleanup/RemoveTestResources.cs]: done

File: retail/interactive-tutorial/TestResourcesSetupCleanup/RemoveTestResources.cs (hunk 15 above)
Comment (mkoshurnikov) [reply retail/interactive-tutorial/TestResourcesSetupCleanup/RemoveTestResources.cs]: done

File: retail/interactive-tutorial/TestResourcesSetupCleanup/CreateTestResources.cs [hunk 16]
```diff
@@ -0,0 +1,342 @@
...
+                break;
+            }
+        }
+
+        return bucketExists;
+    }
+
+    /// <summary>List all existing buckets.</summary>
+    public static IEnumerable<Bucket> ListBuckets() =>
+    storageClient.ListBuckets(projectId);
```
Comment (amanda-tarafa) [inline retail/interactive-tutorial/TestResourcesSetupCleanup/CreateTestResources.cs:103]:
```suggestion
        storageClient.ListBuckets(projectId);
```

File: retail/interactive-tutorial/TestResourcesSetupCleanup/README.md [hunk 17]
```diff
@@ -0,0 +1,44 @@
...
+        - insert user events from `TestResourcesSetupCleanup/resources/user_events.json` to the created events table
+
+```
+$ dotnet run -- CreateTestResourcesTutorial
+```
+
+
+## Remove catalog data
+
+Run the `RemoveTestResources.cs` to perform the following actions:
```
Comment (amanda-tarafa) [inline retail/interactive-tutorial/TestResourcesSetupCleanup/README.md:33]: The RETAIL_ prefix is missing in env vars for all the lines after this one.

**Review by amanda-tarafa — CHANGES_REQUESTED** [review]

File: retail/interactive-tutorial/TestResourcesSetupCleanup/CreateTestResources.cs (hunk 16 above)
Comment (mkoshurnikov) [reply retail/interactive-tutorial/TestResourcesSetupCleanup/CreateTestResources.cs]: done

File: retail/interactive-tutorial/TestResourcesSetupCleanup/README.md (hunk 17 above)
Comment (mkoshurnikov) [reply retail/interactive-tutorial/TestResourcesSetupCleanup/README.md:33]: done

**Review by amanda-tarafa — APPROVED** [review]
LGTM, running tests to check that this all builds. I'll merge if it is green.

**Pull request merged**